zerver/tests/test_docs.py

```diff
@@ -105,20 +106,22 @@ def _test(
                 )
 
     def test_api_doc_endpoints(self) -> None:
-        current_dir = os.path.dirname(os.path.abspath(__file__))
-        api_docs_dir = os.path.join(current_dir, "..", "..", "templates/zerver/api/")
-        files = os.listdir(api_docs_dir)
-
-        def _filter_func(fp: str) -> bool:
-            ignored_files = ["sidebar_index.md", "index.md", "missing.md"]
-            return fp.endswith(".md") and not fp.startswith(".") and fp not in ignored_files
+        api_page_raw = str((self.client_get("/api/").content))
+        ENDPOINT_REGEXP = re.compile(r"\"/api/\s*(.*?)\"")
+        endpoint_list_set = set(re.findall(ENDPOINT_REGEXP, api_page_raw))
 
-        files = list(filter(_filter_func, files))
+        endpoint_list = [f"/api/{endpoint}" for endpoint in endpoint_list_set]
 
-        for f in files:
-            endpoint = f"/api/{os.path.splitext(f)[0]}"
+        for endpoint in endpoint_list:
             self._test(endpoint, "", doc_html_str=True)
 
+        result = self.client_get(
+            "/api/nonexistent-page",
+            follow=True,
+            HTTP_X_REQUESTED_WITH="XMLHttpRequest",
+        )
+        self.assertEqual(result.status_code, 404)
```

Comment:
Can you pull this commit as well as a targeted bug fix for /api/invalid being broken in master into a smaller prep PR that we can merge now?  We're getting 500s in chat.zulip.org from crawlers right now, and I'd like to be able to merge a fix for that (with this test code) tomorrow even if the rest needs more round trips.